### PR TITLE
Add Patreon social link

### LINKED ISSUE #889: Add patreon as a social link
### What version of `starlight` are you using?

0.11.0

### What is your idea?

My idea is to add an option to add a patreon as a social link.

### Why is this feature necessary?

This feature is neccessary as it gives the user a way to know that the project accepts donations from the community.

### Do you have examples of this feature in other projects?

Yes. I remember that Material for Mkdocs has a featyure similiar to this where they accept any social link that has an icon from fontaweome. This allows the user to configure and add any social link and add any icon from fontaweseom to their docs.

The documentation about this feature is here: <https://squidfunk.github.io/mkdocs-material/setup/setting-up-the-footer/?h=social#social-links>

### Participation

- [ ] I am willing to submit a pull request for this issue.

### PR DESCRIPTION
**Pull request by delucis**
#### What kind of changes does this PR include?

- Changes to Starlight code

#### Description

- Closes #889
- Adds a Patreon icon and includes it in our social link config
- N.B. #869 will need updating for this PR

### CODE REVIEW

File: packages/starlight/components/Icons.ts [hunk 1]
```diff
@@ -97,4 +97,6 @@ export const Icons = {
...
 	reddit:
 		'<path d="M14.41 16.87a3.38 3.38 0 0 1-2.37.63 3.37 3.37 0 0 1-2.36-.63 1 1 0 0 0-1.42 1.41 5.11 5.11 0 0 0 3.78 1.22 5.12 5.12 0 0 0 3.78-1.22 1 1 0 1 0-1.41-1.41ZM9.2 15a1 1 0 1 0-1-1 1 1 0 0 0 1 1Zm6-2a1 1 0 1 0 1 1 1 1 0 0 0-1-1Zm7.8-1.22a3.77 3.77 0 0 0-6.8-2.26 16.5 16.5 0 0 0-3.04-.48l.85-5.7 2.09.7a3 3 0 0 0 6-.06v-.02a3.03 3.03 0 0 0-3-2.96 2.98 2.98 0 0 0-2.34 1.16l-3.24-1.1a1 1 0 0 0-1.3.8l-1.09 7.17a16.66 16.66 0 0 0-3.34.49 3.77 3.77 0 0 0-6.22 4.23A4.86 4.86 0 0 0 1 16c0 3.92 4.83 7 11 7s11-3.08 11-7a4.86 4.86 0 0 0-.57-2.25 3.78 3.78 0 0 0 .57-1.97ZM19.1 3a1 1 0 1 1-1 1 1.02 1.02 0 0 1 1-1ZM4.77 10a1.76 1.76 0 0 1 .88.25A9.98 9.98 0 0 0 3 11.92v-.14A1.78 1.78 0 0 1 4.78 10ZM12 21c-4.88 0-9-2.29-9-5s4.12-5 9-5 9 2.29 9 5-4.12 5-9 5Zm8.99-9.08a9.98 9.98 0 0 0-2.65-1.67 1.76 1.76 0 0 1 .88-.25A1.78 1.78 0 0 1 21 11.78l-.01.14Z"/>',
+	patreon:
+		'<path d="M1033 324.4c0-137.8-107.5-250.9-233.6-291.6C643-18 436.6-10.5 287.2 60 106 145.4 49.2 332.6 47 519.3c-1.8 153.5 13.5 557.8 241.6 560.7 169.4 2.1 194.7-216.2 273-321.3 55.8-74.9 127.6-96 216-117.9 152-37.6 255.6-157.5 255.3-316.3z"/>',
```
Comment (HiDeoo) [inline packages/starlight/components/Icons.ts:101]: This looks like huge coordinates, would that render properly in Starlight?

**Comment by delucis** [comment]
@HiDeoo I think it looks OK now? https://deploy-preview-892--astro-starlight.netlify.app/guides/components/#all-icons

**Comment by delucis** [comment]
> Out of curiosity, what tool did you use to scale down the path coordinates?

I tried like 4 different online “SVG resizers” none of which worked and so plopped it in Figma and did it there 😅

**Comment by HiDeoo** [comment]
Ah yeah, that's how I usually do it too, kinda hoped you had a 1-click upload & resize magical tool link ^^ Thanks

**Comment by delucis** [comment]
One must exist somewhere! It’s just an SEO nightmare with loads of marketing for suites of bad online graphics editor tools, so hard to find. Maybe a GitHub search would do better at uncovering something.

Like this, which I needed the other day: https://svg-path-visualizer.netlify.app/

**Pull request merged**